services/utils/__init__.py

```diff
@@ -17,6 +18,12 @@
 log_level = os.environ.get('LOGLEVEL', 'INFO').upper()
 logging.basicConfig(level=log_level)
 
+class InvalidDSNException(Exception):
+    headline = 'Invalid DSN String'
+    def __init__(self, lineno=None):
+        self.message = f"The DSN String Provided is Invalid"
+        self.line_no = lineno
```

Comment:
what is `lineno` for?

Reply:
nit: i think it would be perfectly fine to just `raise Exception("Invalid DSN string")` and not have a separate exception class, as we don't catch this specific exception anywhere anyway

Reply:
Will Fix.

Reply:
`lineno` was by mistake; I will refactor your solution and update the PR.

Reply:
Done.